### PR TITLE
feat: add java files

### PR DESCRIPTION
**Pull request by tzssangglass**
add java files that generated by the FlatBuffers compiler, and the source code for the java of FlatBuffers

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,4 +5,14 @@ lua:
...
-compiled: lua
+
+.PHONY: java
+java:
+	flatc --java ext-plugin.fbs
+	rm -rf java/A6
+	mv A6/ java/
+
+.PHONY: compiled
+
```
Comment (spacewander) [inline Makefile]: Why add two blank line here?

Comment (tzssangglass) [reply Makefile]: my mistake, update

File: Makefile [hunk 2]
```diff
@@ -5,4 +5,13 @@ lua:
...
+.PHONY: java
+java:
+	flatc --java ext-plugin.fbs
+	rm -rf java/io.github.api7.A6
+	mv A6/ java/io.github.api7.A6/
+	find . -name "*.java" | xargs sed -i 's/A6/io.github.api7.A6/g'
+
+.PHONY: compiled
+compiled: lua java
```
Comment (gxthrj) [inline Makefile:17]: Need to add the `java` directory to `.gitignore`.

Comment (tzssangglass) [reply Makefile:17]: I see that the lua code generated by flatbuffers is not in the   `.gitignore`, so I thought the java ones didn't need. @spacewander what do you think?

**Review by nic-chen — COMMENTED** [review]
I think we could use github action to publish jar to the Maven Central Repository automatically
see:
https://docs.github.com/en/actions/guides/publishing-java-packages-with-maven

**Comment by tzssangglass** [comment]
> I think we could use github action to publish jar to the Maven Central Repository automatically
> see:
> https://docs.github.com/en/actions/guides/publishing-java-packages-with-maven

Do it later, because this is the generator and the release version will not be very frequent.

**Review by nic-chen — APPROVED** [review]

**Review by membphis — APPROVED** [review]
missing test  cases, we can add them later

**Pull request merged**